pydantic/_internal/_generate_schema.py

```diff
@@ -1659,10 +1663,15 @@ def _union_is_subclass_schema(self, union_type: Any) -> core_schema.CoreSchema:
     def _subclass_schema(self, type_: Any) -> core_schema.CoreSchema:
         """Generate schema for a Type, e.g. `Type[int]`."""
         type_param = self._get_first_arg_or_any(type_)
+        type_param = _typing_extra.eval_type(type_param, *self._types_namespace)
```

Comment:
Is this one necessary? `_get_first_arg_or_any` already evaluates forward refs

Reply:
This is for annotation like `type['int']`. While `_get_first_arg_or_any` resolves `ForwardRef`, it does not resolve str objects. So `type_param` received here will be `'int'`, at least in some Python version (<= 3.10).

Reply:
I have the same question as @Viicos.

Should we be using `GenerateSchema._type_alias_type_schema` anywhere below instead?

Reply:
The evaluation of forward refs is still a mess, but looking at the other places where `_get_first_arg_or_any` (and `_get_args_resolving_forward_refs`), this is the only place where the extra evaluation is needed. Let's go this way for now but this should be cleaned up at some point.

Reply:
> Should we be using `GenerateSchema._type_alias_type_schema` anywhere below instead?

I don't think `_type_alias_type_schema` is relevant here; we want to get the actual type here, instead of the schema generated from the type.

Reply:
With the changes in `_get_args_resolving_forward_refs`, this can be removed :+1: